lib/mongo/crypt/status.rb

```diff
@@ -84,6 +84,23 @@ def ref
         @status
       end
 
+      # Raises a Mongo::Error:CryptError corresponding to the
+      # information stored in this status
+      #
+      # Does nothing if self.ok? is true
+      def raise_crypt_error
```

Comment:
I'm not sure if there's a better pattern I could use to accomplish this -- maybe 
```
raise Mongo::Error::CryptError.from_status(status)
```
Open to opinions here.